src/WebJobs.Extensions.DurableTask/Listener/TaskActivityShim.cs

```diff
@@ -59,20 +61,24 @@ internal class TaskActivityShim : TaskActivity
             FunctionResult result = await this.executor.TryExecuteAsync(triggerInput, CancellationToken.None);
             if (!result.Succeeded)
             {
+                // Flow the original activity function exception to the orchestration
+                // without the outer FunctionInvocationException.
+                Exception exceptionToReport = StripFunctionInvocationException(result.Exception);
+
                 this.config.TraceHelper.FunctionFailed(
                     this.config.HubName,
                     this.activityName,
                     this.activityVersion,
                     instanceId,
-                    result.Exception?.ToString() ?? string.Empty,
+                    exceptionToReport?.ToString() ?? string.Empty,
                     functionType: FunctionType.Activity,
                     isReplay: false);
 
-                if (result.Exception != null)
+                if (exceptionToReport != null)
                 {
-                    // Preserve the original exception context so that the durable task
-                    // framework can report useful failure information.
-                    ExceptionDispatchInfo.Capture(result.Exception).Throw();
+                    throw new TaskFailureException(
```

Comment:
>throw new TaskFailureException( [](start = 20, length = 31)

We throw `TaskFailureException` here because that is the only exception type where the Durable Task Framework will actually try to deserialize the inner exception back in the orchestrator function.